### PR TITLE
WT-13181 Fix two assert always error messages

### PR DESCRIPTION
**Pull request by quchenhao**

### CODE REVIEW

**Comment by quchenhao** [comment]
@luke-pearson I think we should turn these two assert always messages to asserts. The conditions to fire for these two asserts are two specific and detail to report to the users.

**Comment by luke-pearson** [comment]
WT_ASSERT_ALWAYS isn't really about returning a message to the user but rather guarding against data corruption. If the assert firing in diagnostic was bad enough that it might create data corruption then we should run it in production as well. The message can be vague, the point of the ticket was that the messages were wrong.

**Comment by quchenhao** [comment]
@luke-pearson Can you have a look?

File: src/reconcile/rec_visibility.c [hunk 1]
```diff
@@ -594,18 +594,21 @@ __rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_UPDATE *first_upd
                 continue;
             } else {
                 /*
-                 * For prepared updates written to the date store in salvage, we write the same
-                 * prepared value to the data store. If there is still content for that key left in
-                 * the history store, rollback to stable will bring it back to the data store.
-                 * Otherwise, it removes the key.
+                 * If we are not in eviction, we must be in salvage to reach here. For prepared
+                 * updates written to the date store in salvage, we write the same prepared value to
```
[the remark this answers is not in the document]
Comment (quchenhao) [reply src/reconcile/rec_visibility.c]: I reworded the comment.

**Review by luke-pearson — APPROVED** [review]
LGTM.

**Pull request merged**